tests/test_virtualenv.py

```diff
@@ -47,6 +48,47 @@ def factory(*args, **kwargs):
     return factory
 
 
+@pytest.fixture
+def make_mocked_interpreter_path():
+    def factory(path, sysexec_result):
+        def mock_sysexec(*_):
+            if sysexec_result == RAISE_ERROR:
```

Comment:
First of all, thank you for re-factoring these tests at my request. I know that can be a bit annoying after you've written some useful unit tests.

The core of my re-factoring request was that I was hoping we could ditch `if` statements for distinct test cases. But now there is an `if` statement here and on line 78, so my original hesitation is still there.

Reply:
Not annoying at all! It's an interesting challenge, and, yes, there were absolutely too many `if` statements in the original test.

But thanks to your request, the number of `if` statements got at least cut in half. Now, there are no `if`s in the tests themselves, so their logic should be much clearer. There are only mock/patch-related `if` statements left. Here, I really wanted to avoid copy-pasting the same mock/patch code over several tests. This mock/patch construction unfortunately needs to be a bit complex: a patched `sysfind` returns a mocked `path` to the python interpreter, which has a `__str__`, and whose `sysexec` method returns the version string.
On every one of these steps, something could go wrong and should be tested. The only way I see to test all these cases without conditionals, would be to mock all variants of these situations separately, which, IMO, would make the tests themselves harder to read.

I do get that each fixture creates a new potential source for bugs, and for me, the `if`s that are left are clear and primitive enough that it's worth the tradeoff. I just don't see how I could split this whole thing up into separate `if`-less tests (with or without extra fixtures) that would still be readable and not lead to increasing the amount of lines of test code manyfold. I am open to suggestions on how someone with more experience than me would solve this.